contracts/JBDirectory.sol

```diff
@@ -154,43 +178,31 @@ contract JBDirectory is IJBDirectory, JBOperatable {
     projects = _projects;
   }
 
-  //*********************************************************************//
-  // ---------------------- external transactions ---------------------- //
-  //*********************************************************************//
-
   /**
     @notice
     Update the controller that manages how terminals interact with the ecosystem.
-
     @dev 
     A controller can be set if:
-    - the message sender is the project owner or an operator is changing the controller.
-    - or, the controller hasn't been set yet and the message sender is the controller being set.
-    - or, the current controller is setting a new controller.
-
+    - the message sender is the project owner or an operator having the correct authorization.
+    - or an allowed address is setting a new controller.
     @param _projectId The ID of the project to set a new controller for.
     @param _controller The new controller to set.
   */
-  // TODO(odd-amphora): Revisit access pattern with allowlist.
   function setControllerOf(uint256 _projectId, IJBController _controller)
     external
     override
     requirePermissionAllowingOverride(
       projects.ownerOf(_projectId),
       _projectId,
       JBOperations.SET_CONTROLLER,
-      (address(controllerOf[_projectId]) == address(0) && msg.sender == address(_controller)) ||
-        address(controllerOf[_projectId]) == msg.sender
+      (_setControllerAllowlist[address(_controller)] && _setControllerAllowlist[msg.sender])
```

Comment:
Should we use `isAllowedToSetController` in these cases?

Reply:
isAllowedToSetController is a public function -> would slightly increase gas consumption (alt being a second internal function overloading it, costing then 1U for the jump + deployment overhead, but I guess that's not really necessary here)